### ISSUE TITLE
Rate limit sms urls

### ISSUE THREAD

**Issue by LeeSmet** [itsyouonline/identityserver#678]
Cloudlfare has a facility which can do this - investigate if it gives us enough flexibility

If not, we should deploy a custom middleware which handles this, based on IP address. This is important for the registration page and login page as there the user is not logged in yet but has the ability to send sms messages.

Also, we should store some logging about sending acutal sms messages so a single number can't be spammed from multiple IP's

**Comment by LeeSmet** [comment]
Current cloudlfare plan offers only 1 rate limited url. Upgrading plans will offer 3.

If we upgrade, we could probably manage to fit all the sms sending logic in those 3 urls.

Also not that cf rate limiting periods are not that extensively customizable. Furthermore, some form of rate limiting on logged in urls bases on username is also only possible on our side, so we will need to implement this ourselves either way. 

In conclusion, its probably best to just handle the whole thing ourselves, since this gives more flexible customization options, logging, and anything else we might need.

**Comment by LeeSmet** [comment]
Sms urls have been rate limited based on ip address, which is logged if rate limiting is hit. Rate limiting sending of sms based on phone numbers is to be considered a different issue
